portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,17 +38,33 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+    UserService userService = UserServiceFactory.getUserService();
+
+    String userEmail = "";
```

Comment:
This would be a great place to use an Optional!

Reply:
Done

Reply:
I tried implementing Optional but I get NullPointerException if I dont check that user is logged in

Reply:
I think you could still do this, by having userEmail be an Optional<String>, initialized to Optional.empty(). Then you check to see if the user is logged in, and if they are you set it to Optional.of(userService...)